hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -782,6 +791,32 @@ private List<FileStatusAdapter> listStatusVolume(String volumeStr,
     return res;
   }
 
+  /**
+   * Helper for OFS listStatus
+   * on a bucket to get all snapshots.
+   */
+  private List<FileStatusAdapter> listStatusBucketSnapshot(
+      String volumeName, String bucketName, URI uri) throws IOException {
+    List<OzoneSnapshot> snapshotList =
+        objectStore.listSnapshot(volumeName, bucketName);
+
+    OzoneVolume volume = objectStore.getVolume(volumeName);
+    UserGroupInformation ugi =
+        UserGroupInformation.createRemoteUser(volume.getOwner());
+    String owner = ugi.getShortUserName();
+    String group = getGroupName(ugi);
+
+    OzoneBucket ozoneBucket = getBucket(volumeName, bucketName, false);
+    List<FileStatusAdapter> fileStatusAdapterList = new ArrayList<>();
+    for (OzoneSnapshot ozoneSnapshot : snapshotList) {
+      fileStatusAdapterList.add(
+          getFileStatusAdapterForBucketSnapshot(
+              ozoneBucket, ozoneSnapshot, uri, owner, group));
+    }
+
+    return fileStatusAdapterList;
```

Comment:
```suggestion
    return snapshots.stream()
        .map(ozoneSnapshot -> getFileStatusAdapterForBucketSnapshot(
            ozoneBucket, ozoneSnapshot, uri, owner, group))
        .collect(Collectors.toList());
```
nit: you can use lambdas.